src/supernode/src/main/java/com/dragonflyoss/dragonfly/supernode/service/repair/BaseClientErrorHandler.java

```diff
@@ -0,0 +1,167 @@
+package com.dragonflyoss.dragonfly.supernode.service.repair;
+
+import java.util.concurrent.BlockingDeque;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+
+import com.alibaba.fastjson.JSON;
+
+import com.dragonflyoss.dragonfly.supernode.common.Constants;
+import com.dragonflyoss.dragonfly.supernode.common.domain.ClientErrorInfo;
+import com.dragonflyoss.dragonfly.supernode.common.enumeration.ClientErrorType;
+import com.dragonflyoss.dragonfly.supernode.common.util.ExecutorsUtil;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * The client error will be handled immediately. In the process of handling,
+ * and within 5 seconds after completing handling, the subsequent client errors
+ * with the same taskId and error type will be ignored.
+ *
+ * @author lowzj
+ */
+@Slf4j
+public abstract class BaseClientErrorHandler implements ClientErrorHandler {
+    private static final ExecutorService executor = ExecutorsUtil
+        .newThreadPool("client-error", 20, 200);
+
+    /**
+     * map(taskId, addTime)
+     */
+    private ConcurrentHashMap<String, Integer> running = new ConcurrentHashMap<>();
+
+    private Remover remover = new Remover();
+
+    public BaseClientErrorHandler() {
+        executor.execute(remover);
+    }
+
+    abstract ClientErrorType errorType();
+    abstract void doHandle(final ClientErrorInfo info);
+
+    @Override
+    public boolean isErrorType(ClientErrorType type) {
+        return errorType().equals(type);
+    }
+
+    @Override
+    public void handle(final ClientErrorInfo info) {
+        if (isInvalidInfo(info)) {
+            return;
+        }
+        Integer pre = running.putIfAbsent(info.getTaskId(), currentSeconds());
```

Comment:
one task may have different error.
should we distinguish different error codes?